solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/public-dns.yaml

```diff
@@ -0,0 +1,101 @@
+  visibility: public
+  dnssecConfig:
+    state: "on"
+  cloudLoggingConfig:
+    enableLogging: true
+---
+# NS Record Set for client DNS subzone
+# This will update the client DNS subzone NS record with a specific set of nameservers, configured in setters.
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-public-dns-ns-rset # kpt-set: ${client-name}-standard-public-dns-ns-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: dns.cnrm.cloud.google.com/namespaces/client-name-networking/DNSManagedZone/client-name-standard-public-dns # kpt-set: dns.cnrm.cloud.google.com/namespaces/${client-name}-networking/DNSManagedZone/${client-name}-standard-public-dns
+spec:
+  name: "dns-name" # kpt-set: ${dns-name}
+  type: "NS"
+  ttl: 21600
+  managedZoneRef:
+    name: client-name-standard-public-dns # kpt-set: ${client-name}-standard-public-dns
+    namespace: client-name-networking # kpt-set: ${client-name}-networking
+  rrdatas: # kpt-set: ${dns-nameservers}
+    - "ns-cloud-a1.googledomains.com."
+    - "ns-cloud-a2.googledomains.com."
+    - "ns-cloud-a3.googledomains.com."
+    - "ns-cloud-a4.googledomains.com."
+---
+# SOA Record Set for Client DNS subzone
+# This will update the client DNS subzone SOA's primary name server, configured in setters.
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-public-dns-soa-rset # kpt-set: ${client-name}-standard-public-dns-soa-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: dns.cnrm.cloud.google.com/namespaces/client-name-networking/DNSManagedZone/client-name-standard-public-dns # kpt-set: dns.cnrm.cloud.google.com/namespaces/${client-name}-networking/DNSManagedZone/${client-name}-standard-public-dns
+spec:
+  name: "dns-name" # kpt-set: ${dns-name}
+  type: "SOA"
+  ttl: 21600
+  managedZoneRef:
+    name: client-name-standard-public-dns # kpt-set: ${client-name}-standard-public-dns
+    namespace: client-name-networking # kpt-set: ${client-name}-networking
+  rrdatas: # kpt-set: ${dns-primary-nameserver}
+    - "ns-cloud-a1.googledomains.com. cloud-dns-hostmaster.google.com. 1 21600 3600 259200 300"
+---
+# Record Set for client DNS subzone in public core DNS zone
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-core-public-dns-ns-rset # kpt-set: ${client-name}-standard-core-public-dns-ns-rset
+  namespace: networking # kpt-set: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: dns-project-id # kpt-set: ${dns-project-id}
+spec:
```

Comment:
I think you can put a depends on :
dns.cnrm.cloud.google.com/namespaces/client-name-networking/DNSManagedZone/client-name-standard-public-dns # kpt-set: dns.cnrm.cloud.google.com/namespaces/${client-name}-networking/DNSManagedZone/${client-name}-standard-public-dns

Reply:
I had to remove this depends on since the ManagedZoneRef is from the tier1 root sync and this resource is part of the tier2 root sync.

Reply:
you need to set a depends on otherwise this resource will be  included in the config sync apply group 0 and this breaks the current depends on hierarchy. the depends-on I suggested is a resource inside this package and maintains a functionnal depends-on hierarchy

Reply:
@cartyc is investigating a better depends-on solution to simplify our current very complex hierarchy of apply groups

Reply:
Fixed/Added